src/metabase/api/notification.clj

```diff
@@ -16,6 +19,41 @@
       api/check-404
       models.notification/hydrate-notification))
 
+(api/defendpoint GET "/"
+  "List notifications.
+  - `creator_id`: if provided returns only notifications created by this user
+  - `recipient_id`: if provided returns only notification that has recipient_id as a recipient
+  - `card_id`: if provided returns only notification that has card_id as payload"
+  [creator_id recipient_id card_id include_inactive]
+  {creator_id       [:maybe ms/PositiveInt]
+   recipient_id     [:maybe ms/PositiveInt]
+   card_id          [:maybe ms/PositiveInt]
+   include_inactive [:maybe ms/BooleanValue]}
+  (->> (t2/select :model/Notification
```

Comment:
Given that we restrict by permissions in-memory, I think we need a reducible select so that this can scale.